support-services/src/test/scala/com/gu/support/redemption/GetCodeStatusSpec.scala

```diff
@@ -0,0 +1,77 @@
+package com.gu.support.redemption
+
+import com.gu.support.redemption.GetCodeStatus.{CodeAlreadyUsed, NoSuchCode}
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.Future
+
+class GetCodeStatusSpec extends AsyncFlatSpec with Matchers {
+
+  "getCodeStatus" should "handle an available code" in {
+    val getCodeStatus = GetCodeStatus.withDynamoLookup {
+      case "CODE" => Future.successful(Some(Map("available" -> true)))
+    }
+    getCodeStatus(RedemptionCode("CODE")).map {
+      _ should be(Right(()))
+    }
+  }
+
+  it should "handle an available code with extra attributes lying around" in {
+    val getCodeStatus = GetCodeStatus.withDynamoLookup {
+      case "CODE" => Future.successful(Some(Map(
+        "available" -> true,
+        "otherFields" -> false
+      )))
+    }
+    getCodeStatus(RedemptionCode("CODE")).map {
+      _ should be(Right(()))
+    }
+  }
+
+  it should "handle an NON available code" in {
+    val getCodeStatus = GetCodeStatus.withDynamoLookup {
+      case "CODE" => Future.successful(Some(Map("available" -> false)))
+    }
+    getCodeStatus(RedemptionCode("CODE")).map {
+      _ should be(Left(CodeAlreadyUsed))
+    }
+  }
+
+  it should "handle an NON EXISTENT code" in {
+    val getCodeStatus = GetCodeStatus.withDynamoLookup {
+      case "CODE" => Future.successful(None)
+    }
+    getCodeStatus(RedemptionCode("CODE")).map {
+      _ should be(Left(NoSuchCode))
+    }
+  }
+
+//  it should "handle an code with invalid attrbibute type " in {
+//    val getCodeStatus = GetCodeStatus.withDynamoLookup {
+//      case "CODE" => Future.successful(Some(Map("available" -> new AttributeValue().withS("true"))))
+//    }
+//    recoverToSucceededIf[RuntimeException] {
+//      getCodeStatus(RedemptionCode("CODE"))
+//    }
+//  }
```

Comment:
this test would have to happen against DynamoTableAsync further up in this PR, as the code has been pushed down to simplify the API.  It could be worth extracting the non-side effect parts of that file, and testing them? the tests would be messy but wouldn't be mixed with any business logic tests such as the current one.
https://github.com/guardian/support-frontend/blob/79d715962389a72a0c7bdb10ee6efa055297544a/support-services/src/main/scala/com/gu/support/redemption/DynamoTableAsync.scala#L63